src/lib/server/server-provider.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  InjectionToken,    // tslint:disable-line:no-unused-variable
+  ComponentRef,      // tslint:disable-line:no-unused-variable
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {BEFORE_APP_SERIALIZED} from '@angular/platform-server';
+
+import {
+  BreakPoint,
+  BREAKPOINTS,
+  CLASS_NAME,
+  MatchMedia,
+  ServerStylesheet,
+  SERVER_TOKEN,
+} from '@angular/flex-layout';
+
+let UNIQUE_CLASS = 0;
+const DEBUG_FLAG = false;
+
+/**
+ * create @media queries based on a virtual stylesheet
+ * * Adds a unique class to each element and stores it
+ *   in a shared classMap for later reuse
+ */
+function formatStyle(stylesheet: Map<HTMLElement, Map<string, string|number>>,
+                     _document: Document,
+                     mediaQuery: string,
+                     classMap: Map<HTMLElement, string>) {
+  let styleText = DEBUG_FLAG ? `
```

Comment:
Rather than formatting the css manually, it would probably be better to use some existing node css formatter / prettifier (still only when debugging). As it is now, the method is pretty hard to mentally parse.

Reply:
Do you have a CSS formatter in mind? Maybe this can be done in a follow-up PR?